src/types.ts

```diff
@@ -21,10 +21,10 @@ export interface MirrorOptions {
    */
   customFilename?: string;
   /**
-   * Whether to download from the base URL only,
-   * ignoring customDir and customFilename
+   * A function allowing customization of the url returned
+   * from getArtifactRemoteURL().
    */
-  baseOnly?: boolean;
+  resolveAssetURL?: (opts: Omit<MirrorOptions, 'resolveAssetURL'>) => Promise<string>;
```

Comment:
We should probably give it the full options object tbh.  The URL resolver will need to know platform / version / arch, etc.